lib/time.ml

```diff
@@ -52,15 +52,17 @@ module SleepQueue =
   Binary_heap.Make (struct
                      type t = sleep
                      let compare { time = t1; _ } { time = t2; _ } =
-                       (* bheap library only operates on maximums, but we
-                          need the minimum, so invert the order *)
-                       let invert_comparison x = -x in
-                       invert_comparison (compare t1 t2)
+                       compare t1 t2
                    end)
 
 (* Threads waiting for a timeout to expire: *)
-(* bheap datastructure needs to be created with initial strictly positive capacity *)
-let sleep_queue = SleepQueue.create 1
+let sleep_queue =
+  let dummy = {
+    time = Monotonic.time ();
+    canceled = false;
+    thread = Lwt.wait () |> snd; }
```

Comment:
I'm not sure what the consequences of this actually is for the `lwt` scheduler, so maybe you want to provide another `Lwt.u` value here instead.

Reply:
So if you can assert that `dummy` is not really used and must not returned by `bheap`, it seems fine.

Reply:
Indeed, it's only used to fill the internal array without keeping references to values that could otherwise be collected.